modules/ServerAPI/src/client/Experiment.coffee

```diff
@@ -781,4 +788,17 @@ class ExperimentBaseController extends BaseEntityController
 
 	displayInReadOnlyMode: =>
 		super()
-		@$('.bv_openInQueryToolWrapper').hide()
\ No newline at end of file
+		@$('.bv_openInQueryToolWrapper').hide()
+	
+	setupEndpointsController: =>
+		# Hack to get Protocol working with Thing-based ACASFormStateTable classes
+		@model.lsProperties = {'defaultValues': []}
+		# The 'data column order' states are a StateTable
+		# Get any existing states with that type & kind
+		endpointStates = @model.get("lsStates").getStatesByTypeAndKind "metadata", "data column order"
+		# Create the controller for the Endpoints table which manages all the states
+		@endpointListController = new EndpointListController
+			el: @$('.bv_endpointTable')
+			collection: endpointStates
+			model: @model
+			readOnly: true
```

Comment:
Code for rendering the read-only Endpoint Table in the experiment browser and experiment editor.